packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanBar.tsx

```diff
@@ -54,6 +54,52 @@ function toPercentInDecimal(value: number) {
   return `${value * 100}%`;
 }
 
+function SpanBarCriticalPath(props: { criticalPathViewStart: number; criticalPathViewEnd: number }) {
+  const [shouldLoadTooltip, setShouldLoadTooltip] = useState(false);
+
+  // Load tooltip only when hovering over critical path segment
+  // to reduce initial load time of trace page by ~300ms for 500 spans
+  if (shouldLoadTooltip) {
+    return (
+      <Tooltip
+        placement="top"
+        // defaultOpen is needed to show the tooltip when shouldLoadTooltip changes to true
+        defaultOpen
+        title={
+          <div>
+            A segment on the <em>critical path</em> of the overall trace/request/workflow.
+          </div>
+        }
+      >
+        <div
```

Comment:
can we avoid duplicating this code segment?

Reply:
Yes, see https://github.com/jaegertracing/jaeger-ui/pull/2718#discussion_r2041031917

I will change this tomorrow